PRINCIPLES.md

```diff
@@ -0,0 +1,254 @@
+
+2. [**The principle of immutable configuration versions**](#2-immutable-configuration-versions)
+
+    _Desired State_ is stored in a way that supports versioning, immutability of versions, and retains a complete version history.
+    We call systems that store desired state in this way _State Stores_.
+
+3. [**The principle of continuous state reconciliation**](#3-continuous-state-reconciliation)
+
+    Software agents continuously, and automatically, compare a system's _Actual State_ to its _Desired State_.
+    If the actual and desired states differ, automated actions are immediately attempted to reconcile them.
+    These differences could be due to the actual state drifting from the desired state, or the desired state changing intentionally.
+
+4. [**The principle of operations through declaration**](#4-operations-through-declaration)
+
+    When wishing to operate on a software system, a human or software agent will not interact with the running system and modify it directly.
+    Instead, the agent will create a new declarative version of the desired state in the state store.
+
+## Introduction
+
+The software systems that we manage vary widely; from battery powered embedded systems driven by microcontrollers, to globally distributed applications with millions of users. 
+Most systems are composed of subsystems themselves made of software.
+It is impossible to comprehensively outline all the specific practices for managing such a variety of systems.
+
+However, despite the myriad differences, several important principles emerge that simplify the task of reliably managing and operating _all_ software systems at scale. GitOps attempts to capture some of these principles in a coherent framework.
+The GitOps principles can be applied to managing entire software systems or applied only to parts of larger systems.
+
+See also the [Rationale for GitOps](RATIONALE.md).
+
+## Scope
+
+GitOps concerns the verifiable behaviour of computer systems and their interfaces.
+Specifically, GitOps is _not_ about human processes, and is not intended as a model for judging human organisational designs and operational practices.
+
+The GitOps principles are to be used as guiding principles in the development of modern software and system operations. They do not form a concrete specification.
+
+The GitOps principles are a direction, not a destination. They should be applied pragmatically. For example, whilst desirable to apply them strictly to an entire software systems, they can also be applied loosely to selected parts of larger systems as part of a progressive adoption.
+
+## The GitOps Principles
+
+### 1. Declarative configuration
+
+<div style="background: #E3F2FD; border-left: 3px solid #0D47A1; padding:10px; margin-bottom:5px; font-style:italic">
+A system managed by GitOps must have its <em>Desired State</em> expressed declaritively as data.
+This declarative data must be in a format writable and readable by both humans and software.
```

Comment:
I explicitly lay this out separately because I want to emphasise the fact that the definition is _data_, not code. I know "declarative" does this already, and saying "declarative data" is somewhat redundant, but it's not clear with everyone in practice and (Homoiconic languages aside), this is a big distinction.

Reply:
The language as is feels stilted though. Let me try to fix that.